File: pinyougou-parent/pinyougou-manager-web/src/main/java/com/pinyougou/manager/controller/BrandController.java
package com.pinyougou.manager.controller;

import com.alibaba.dubbo.config.annotation.Reference;
import com.pinyougou.pojo.TbBrand;
import com.pinyougou.sellergoods.service.BrandService;
import entity.PageResult;
import entity.Result;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.ArrayList;
import java.util.List;
import java.util.Map;

/**
 * Created by crowndint on 2018/10/13.
 */
@RestController
@RequestMapping("brand")
public class BrandController {

    @Reference
    private BrandService brandService;

    @RequestMapping("findAll")
    public List<TbBrand> finDAll() {

        return brandService.findAll();
    }

    @RequestMapping("findByPage")
    public PageResult findByPage(int pageNum, int pageSize) {

        return brandService.findByPage(pageNum, pageSize);
    }

    @RequestMapping("searchByPage")
    public PageResult searchByPage(@RequestBody TbBrand brand, int pageNum, int pageSize) {

        return brandService.findByPage(brand, pageNum, pageSize);
    }

    @RequestMapping("findOne")
    public TbBrand findOne(Long id) {

        return brandService.findOne(id);
    }

    @RequestMapping("update")
    public Result update(@RequestBody TbBrand brand) {
        Result result = new Result();
        try {
            brandService.update(brand);
            result.setSuccess(true);
            result.setMessage("添加成功");
        } catch (Exception e) {
            e.printStackTrace();
            result.setSuccess(false);
            result.setMessage("添加失败");
        }
        return result;
    }

    @RequestMapping("delete")
    public Result delete(Long ids[]) {
        Result result = new Result();
        try {
            brandService.delete(ids);
            result.setSuccess(true);
            result.setMessage("删除成功");
        } catch (Exception e) {
            e.printStackTrace();
            result.setSuccess(false);
            result.setMessage("删除失败");
        }
        return result;
    }

    @RequestMapping("selectOptionsList")
    public List<Map> selectOptionsList() {

        return brandService.selectOptionsList();
    }


}

File: pinyougou-parent/pinyougou-manager-web/src/main/webapp/js/controller/brandController.js
app.controller('brandController', function ($scope,$controller,brandService) {

    //继承
    $controller('baseController', {$scope: $scope});

    $scope.findAll=function () {

        brandService.findAll().success(function (response) {
            $scope.list=response;
        });
    }



    $scope.findPage=function (pageNum, pageSize) {

        brandService.findPage(pageNum, pageSize).success(function (response) {
            //响应数据回来，数据模型改变了，视图也会改变
            $scope.paginationConf.totalItems=response.total;
            $scope.list= response.rows;
        });
    }

    //必须定义
    $scope.searchEntity = {}

    $scope.search=function (pageNum, pageSize) {

        brandService.searchByPage(pageNum, pageSize, $scope.searchEntity).success(function (response) {

            $scope.paginationConf.totalItems=response.total;
            $scope.list=response.rows;
        });
    }

    $scope.pojo = {};
    $scope.update=function () {

        brandService.update($scope.pojo).success(function (response) {
            if (response.success) {
                $scope.reloadList();
            } else {
                alert(response.message);
            }
        });
    }

    $scope.findOne=function (id) {

        brandService.findOne(id).success(function (response) {
            $scope.pojo=response;
        });
    }

    $scope.delete=function () {

        brandService.delete($scope.selectIds).success(function (response) {

            if (response.success) {
                $scope.reloadList();
            }else {
                alert(response.message);
            }
        });
    }



});